PetApi/Controllers/PetStoreController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using PetApiTest;
+
+namespace PetApi.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class PetStoreController : ControllerBase
+    {
+        private static List<Pet> pets = new List<Pet>();
+
+        [HttpPost("addNewPet")]
+        public Pet AddPet(Pet pet)
+        {
+            pets.Add(pet);
+            return pet;
+        }
+
+        [HttpGet("pets")]
+        public IEnumerable<Pet> GetAllPets()
+        {
+            return pets;
+        }
+
+        [HttpGet("getByType/{type}")]
+        public List<Pet> GetPetsByType(string type)
+        {
+            return pets.Where(pet => pet.Type == type).ToList();
+        }
+
+        [HttpGet("getByColor/{color}")]
+        public List<Pet> GetPetsByColor(string color)
+        {
+            return pets.Where(pet => pet.Color == color).ToList();
+        }
+
+        [HttpGet("getOnePet/{name}")]
+        public Pet GetPetByName(string name)
```

Comment:
Why always get pet by index 0

Reply:
Yes, just now I saw that "FirstorDefault can be better"